packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit_api_impls.dart

```diff
@@ -154,6 +164,48 @@ Iterable<WKAudiovisualMediaTypeEnumData> _toWKAudiovisualMediaTypeEnumData(
   });
 }
 
+extension _NavigationActionDataConverter on WKNavigationActionData {
+  WKNavigationAction toNavigationAction() {
+    return WKNavigationAction(
+      request: request.toNSUrlRequest(),
+      targetFrame: targetFrame.toWKFrameInfo(),
+    );
+  }
+}
+
+extension _WKFrameInfoDataConverter on WKFrameInfoData {
+  WKFrameInfo toWKFrameInfo() {
+    return WKFrameInfo(isMainFrame: isMainFrame);
+  }
+}
+
+extension _NSUrlRequestDataConverter on NSUrlRequestData {
+  NSUrlRequest toNSUrlRequest() {
+    return NSUrlRequest(
+      url: url,
+      httpBody: httpBody,
+      httpMethod: httpMethod,
+      allHttpHeaderFields: allHttpHeaderFields.cast(),
+    );
+  }
+}
+
+extension _WKNSErrorDataConverter on NSErrorData {
+  NSError toNSError() {
+    return NSError(
+      domain: domain,
+      code: code,
+      localizedDescription: localiziedDescription,
+    );
+  }
+}
+
+extension _WKScriptMessageDataConverter on WKScriptMessageData {
+  WKScriptMessage toWKScriptMessage() {
+    return WKScriptMessage(name: name, body: body);
+  }
+}
```

Comment:
I agree. I think we can use the enums and data classes generated in from pigeon once https://github.com/flutter/flutter/issues/53191 lands. This would save us alot of code.